src/utils/values.c

```diff
@@ -840,3 +840,80 @@ sr_val_to_buff(const sr_val_t *value, char buffer[], size_t size)
 
     return len;
 }
+
+API int
+sr_tree_to_val(const struct lyd_node *data, const char *path, sr_val_t **value)
+{
+    sr_error_info_t *err_info = NULL;
+    struct ly_set *set = NULL;
+
+    SR_CHECK_ARG_APIRET(!data || !path || !value, NULL, err_info)
+
+    *value = NULL;
+
+    set = lyd_find_path(data, path);
+
+    if (!set) {
+        sr_errinfo_new_ly(&err_info, lyd_node_module(data)->ctx);
+        goto cleanup;
+    } else if (!set->number) {
+        /* Not building err_info to avoid error logs when no item found */
+        ly_set_free(set);
+        return SR_ERR_NOT_FOUND;
+    } else if (set->number > 1) {
+        sr_errinfo_new(&err_info, SR_ERR_INVAL_ARG, NULL, "More subtrees match \"%s\".", path);
+        goto cleanup;
+    }
+
+    /* create return value */
+    *value = malloc(sizeof **value);
+    SR_CHECK_MEM_GOTO(!*value, err_info, cleanup);
+
+    if ((err_info = sr_val_ly2sr(set->set.d[0], *value))) {
+        goto cleanup;
+    }
+
+    /* success */
+cleanup:
+    ly_set_free(set);
+    return sr_api_ret(NULL, err_info);
+}
+
+API int
+sr_tree_to_values(const struct lyd_node *data, const char *xpath, sr_val_t **values, size_t *value_cnt)
+{
+    sr_error_info_t *err_info = NULL;
+    struct ly_set *set = NULL;
+    uint32_t i;
+
+    SR_CHECK_ARG_APIRET(!data || !xpath || !values || !value_cnt, NULL, err_info);
+
+    *values = NULL;
+    *value_cnt = 0;
+
+    set = lyd_find_path(data, xpath);
+
+    if (set) {
```

Comment:
Should probably be handled similarly as in the previous function, explicit check and error.

Reply:
Done please check